### ISSUE TITLE
Impossible to switch sessions without tscon.exe

### ISSUE THREAD

**Issue by alexchandel** [stascorp/rdpwrap#129]
On older versions of windows like Windows 7 Home Premium, rdpwrap succeeds in creating remote sessions, but there's no way to switch between multiple sessions that already exists. This can lead to the gradual accumulation of multiple sessions when multiple sign-ons occur.

**Comment by binarymaster** [comment]
What do you suggest? It isn't so easy to add `tscon.exe` to the installer package, because it is different between Windows builds.

**Comment by alexchandel** [comment]
I'm not sure of an immediate solution to this problem. I've tried moving tscon's around between different versions and they're pretty incompatible.

Do you know how `tscon.exe` actually accomplishes changing the session the display is attached to? It works from the console session, so it must be somewhat generic.

It might be necessary to have a separate tool that can perform session changing to solve this.

**Comment by binarymaster** [comment]
>Do you know how `tscon.exe` actually accomplishes changing the session the display is attached to? It works from the console session, so it must be somewhat generic.

Yes, it uses `WinStationConnect` from `winsta.dll`. It's undocumented function though the headers can be found on the Internet.

http://www.remkoweijnen.nl/blog/2008/11/26/executing-a-fast-user-switch-programmatically-part-1/

>It might be necessary to have a separate tool that can perform session changing to solve this.

Windows allows to choose the session on RDP connect. At least Windows Server editions does support that:

![image](https://cloud.githubusercontent.com/assets/578406/17049715/961268a6-4ff7-11e6-99ed-db4c00100e17.png)

**Comment by AXEBro** [comment]
@alexchandel  - if you run the RDPConf.exe file, there's an option to limit single sign-on per user. I believe that's what you are asking for ..

**Comment by AXEBro** [comment]
![image](https://cloud.githubusercontent.com/assets/14988648/17325487/ddcb8e72-5879-11e6-9a7a-85b60ff0c186.png)

**Comment by alexchandel** [comment]
@AXEBro No I want the possibility for multiple sessions. I just want to be able to switch between them like `tscon.exe` does.

**Comment by alexchandel** [comment]
@binarymaster How exactly does it call this? I tried to write a tool that does this, but it doesn't work.

After RDP'ing in, assuming my remote session's ID is 2, the desired session is 1, and both sessions have the same user, I'm effectively doing this:

```C
char pword = "";
WinStationConnectA(NULL, 1, 2, pword, false);
```

However this fails, and `GetLastError()` returns "1", which is `ERROR_INVALID_FUNCTION`, which makes no sense.

**Comment by binarymaster** [comment]
@alexchandel The first argument isn't a pointer but a handle. Also you may try this:

```c
#define WTS_CURRENT_SERVER_HANDLE 0

WinStationConnectW(WTS_CURRENT_SERVER_HANDLE, 1, 2, NULL, false);
```
